### PR TITLE
test_custom_4d_attention_mask skip with sliding window attn

### PR DESCRIPTION
**Pull request by poedator**
This is a follow-up to #30348 
`tests/models/qwen2/test_modeling_qwen2.py::Qwen2ModelTest::test_custom_4d_attention_mask` was failing because of use of sliding window attention. 
While this sliding window attention does not preclude use of custom 4D masks, it requires modification of the original `test_custom_4d_attention_mask`.

Summoning @ArthurZucker @gante to review

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Sounds good, cc @ydshieh will help you 😉

**Review by ArthurZucker — APPROVED** [review]
